### ISSUE TITLE
[Event Parameter Request] codeunit 5626 "FA General Report" More Parameters in OnBeforeCalcFAPostedAmount

### ISSUE THREAD

**Issue by matthias-planzer** [microsoft/ALAppExtensions#18338]
Could you please implement this Event Param Request? Thanks?
```diff --git a/src/FAGeneralReport.Codeunit.al b/src/FAGeneralReport.Codeunit.al
index 94cfe06..1513eaf 100644
--- a/src/FAGeneralReport.Codeunit.al
+++ b/src/FAGeneralReport.Codeunit.al
@@ -82,6 +82,9 @@ codeunit 5626 "FA General Report"
     end;
 
     procedure CalcFAPostedAmount(FANo: Code[20]; PostingType: Integer; Period: Option "Before Starting Date","Net Change","at Ending Date"; StartingDate: Date; EndingDate: Date; DeprBookCode: Code[10]; BeforeAmount: Decimal; UntilAmount: Decimal; OnlyReclassified: Boolean; OnlyBookValue: Boolean): Decimal
+    var
+        Amount: Decimal;
+        IsHandled: Boolean;
     begin
         if PostingType = 0 then
             exit(0);
@@ -134,7 +137,9 @@ codeunit 5626 "FA General Report"
                 FADeprBook.FieldNo("Depreciable Basis"):
                     SetRange("Part of Depreciable Basis", true);
             end;
-            OnBeforeCalcFAPostedAmount(FALedgEntry, PostingType);
+            OnBeforeCalcFAPostedAmount(FALedgEntry, PostingType, Period, BeforeAmount, UntilAmount, Amount, IsHandled);
+            if IsHandled then
+                exit(Amount);
             case Period of
                 Period::"Before Starting Date":
                     SetRange("FA Posting Date", 0D, StartingDate - 1);
@@ -278,7 +283,7 @@ codeunit 5626 "FA General Report"
     end;
 
     [IntegrationEvent(false, false)]
-    local procedure OnBeforeCalcFAPostedAmount(var FALedgerEntry: Record "FA Ledger Entry"; PostingType: Integer)
+    local procedure OnBeforeCalcFAPostedAmount(var FALedgerEntry: Record "FA Ledger Entry"; PostingType: Integer; Period: Option "Before Starting Date","Net Change","at Ending Date"; BeforeAmount: Decimal; UntilAmount: Decimal; var Amount: Decimal; var IsHandled: Boolean)
     begin
     end;
 
```

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 42200.
